tests/suites/test_suite_gcm.function

```diff
@@ -16,15 +16,19 @@ static int check_multipart( mbedtls_gcm_context *ctx,
     int ok = 0;
     uint8_t *output = NULL;
     size_t n2 = input->len - n1;
+    size_t n1_add = n1 < add->len ? add->len - n1 : add->len;
```

Comment:
I am not a big fan of this correlation of the split of additional data and data to encrypt/decrypt. Please add another argument for the split of additionnal data and in the caller functions change to two nested for loops.